spec/app/ics-101-interchain-swap/README.md

```diff
@@ -0,0 +1,1613 @@
+      const swapResEncoded = protobuf.encode(swapRes);
+      return swapResEncoded;
+
+    default:
+      return;
+  }
+}
+```
+
+`onAcknowledgePacket` is called by the routing module when a packet sent by this module has been acknowledged.
+
+```ts
+// OnAcknowledgementPacket implements the IBCModule interface
+
+// OnAcknowledgementPacket processes the packet acknowledgement and performs actions based on the acknowledgement type
+function OnAcknowledgementPacket(packet: Packet, data: IBCSwapPacketData, ack: Acknowledgement) {
+  switch (ack.response.type) {
+    case "Acknowledgement_Error":
+      return store.refundPacketToken(packet, data);
+    default:
+      switch (data.type) {
+        case "MAKE_POOL":
+          const msgMakePool: MsgMakePoolRequest = protobuf.decode(data.Data);
+          const errMakePool = store.OnMakePoolAcknowledged(msgMakePool, data.StateChange.PoolId);
+          abortTransactionUnless(errMakePool === undefined);
+          break;
+
+        case "TAKE_POOL":
+          const msgTakePool: MsgTakePoolRequest = protobuf.decode(data.Data);
+          const errTakePool = store.OnTakePoolAcknowledged(msgTakePool);
+          abortTransactionUnless(errTakePool === undefined);
+          break;
+
+        case "SINGLE_DEPOSIT":
+          const msgSingleDeposit: MsgSingleAssetDepositRequest = protobuf.decode(data.Data);
+          const resSingleDeposit: MsgSingleAssetDepositResponse = protobuf.decode(ack.GetResult());
+          const errSingleDeposit = store.OnSingleAssetDepositAcknowledged(msgSingleDeposit, resSingleDeposit);
+          abortTransactionUnless(errSingleDeposit === undefined);
+          break;
+
+        case "MAKE_MULTI_DEPOSIT":
+          const msgMakeMultiDeposit: MsgMakeMultiAssetDepositRequest = protobuf.decode(data.Data);
+          const resMakeMultiDeposit: MsgMultiAssetDepositResponse = protobuf.decode(ack.GetResult());
+          const errMakeMultiDeposit = store.OnMakeMultiAssetDepositAcknowledged(
+            msgMakeMultiDeposit,
+            resMakeMultiDeposit
+          );
+          abortTransactionUnless(errMakeMultiDeposit === undefined);
+          break;
+
+        case "TAKE_MULTI_DEPOSIT":
+          const msgTakeMultiDeposit: MsgTakeMultiAssetDepositRequest = protobuf.decode(data.Data);
+          const resTakeMultiDeposit: MsgTakePoolResponse = protobuf.decode(ack.GetResult());
+          const errTakeMultiDeposit = store.OnTakeMultiAssetDepositAcknowledged(msgTakeMultiDeposit, data.StateChange);
+          abortTransactionUnless(errTakeMultiDeposit === undefined);
+          break;
+
+        case "MULTI_WITHDRAW":
+          const msgMultiWithdraw: MsgMultiAssetWithdrawRequest = protobuf.decode(data.Data);
+          const resMultiWithdraw: MsgMultiAssetWithdrawResponse = protobuf.decode(ack.GetResult());
+          const errMultiWithdraw = store.OnMultiAssetWithdrawAcknowledged(msgMultiWithdraw, resMultiWithdraw);
```

Comment:
I think we're missing a definition for `OnMultiAssetWithdrawAcknowledged` function.